nvm.sh

```diff
@@ -8,6 +8,11 @@
 
 { # this ensures the entire script is downloaded #
 
+# Unload any existing environment variables
+if type nvm | grep -i function > /dev/null; then
```

Comment:
This would have to also `2>&1` in case there's stderr output. Also, does `function` appear outside of bash? Also, please do `command grep`.

Any reason not to do this after `nvm_has` is created, to use that?